src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php

```diff
@@ -151,48 +151,53 @@ private function getThumbnailImage(CategoryId $categoryId)
         $image = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '.jpg';
         $imageTypes = ImageType::getImagesTypes('categories');
 
-        $thumb = '';
-        $imageTag = '';
-        $formattedSmall = ImageType::getFormattedName('small');
-        foreach ($imageTypes as $k => $imageType) {
-            if ($formattedSmall == $imageType['name']) {
-                $thumb = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '-' . $imageType['name'] . '.jpg';
-                if (is_file($thumb)) {
-                    $imageTag = ImageManager::thumbnail(
-                        $thumb,
-                        'category_' . (int) $categoryId->getValue() . '-thumb.jpg',
-                        (int) $imageType['width'],
-                        'jpg',
-                        true,
-                        true
-                    );
+        if (count($imageTypes) > 0) {
+            $thumb = '';
+            $imageTag = '';
+            $formattedSmall = ImageType::getFormattedName('small');
+            $imageType = new ImageType();
+            foreach ($imageTypes as $k => $imageType) {
+                if ($formattedSmall == $imageType['name']) {
+                    $thumb = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '-' . $imageType['name'] . '.jpg';
```

Comment:
I see you stumbled upon this code we are not proud of 😄.
Unfortunately we kept the previous behavior which is a bit raw, but we might be able to improve it in the future.

Reply:
Thanks for pointing this out, resolved this stumbling in a different way in Behat tests in the next pull request.